srai/models/gtfs2vec_model.py

```diff
@@ -41,29 +51,33 @@ def __init__(
             nn.Linear(n_embed, n_hidden), nn.ReLU(), nn.Linear(n_hidden, n_features)
         )
 
-    def forward(self, x: torch.Tensor) -> torch.Tensor:
+    def forward(self, x: "torch.Tensor") -> "torch.Tensor":
         """
         Forward pass.
 
         Args:
             x (torch.Tensor): Input tensor.
         """
-        embedding: torch.Tensor = self.encoder(x)
+        embedding: "torch.Tensor" = self.encoder(x)
         return embedding
 
-    def configure_optimizers(self) -> torch.optim.Optimizer:
+    def configure_optimizers(self) -> "torch.optim.Optimizer":
         """Configure optimizer."""
+        import torch
+
         optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
         return optimizer
 
-    def training_step(self, batch: torch.Tensor, batch_idx: Any) -> torch.Tensor:
+    def training_step(self, batch: "torch.Tensor", batch_idx: Any) -> "torch.Tensor":
         """
         Training step.
 
         Args:
             batch (torch.Tensor): Batch.
             batch_idx (Any): Batch index.
         """
+        from torch.nn import functional as F
```

Comment:
Will this import run only once? If not, maybe having it in a training step will hurt performance.

Reply:
Based on this answer https://stackoverflow.com/a/37077911/7766101 it should run only once and not add significant overhead time.

Reply:
Ok, fine then ;)